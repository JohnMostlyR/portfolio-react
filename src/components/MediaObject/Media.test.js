import React from 'react';
import {shallow} from 'enzyme';

import Media from './index';
import MediaImage from './MediaImage';
import MediaBody from './MediaBody';

describe('Media', () => {
  const IMAGE_SOURCE = '/test.png';
  const IMAGE_ALT = 'test';

  let wrapper;
  let mediaImageProps = {};
  let mediaBodyProps = {};

  beforeEach(() => {
    wrapper = shallow(<Media imageAlt={IMAGE_ALT} imageSource={IMAGE_SOURCE}/>);
  });

  it('should render and match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have an MediaImage component', () => {
    const mediaImage = wrapper.find(MediaImage);
    expect(mediaImage).toHaveLength(1);
  });

  it('that should receive the required props', () => {
    const mediaImage = wrapper.find(MediaImage);
    mediaImageProps = mediaImage.props();
    expect(mediaImage.prop('imageSource')).toBe(IMAGE_SOURCE);
    expect(mediaImage.prop('imageAlt')).toBe(IMAGE_ALT);
  });

  it('and receive the optional props', () => {
    const optionalProps = {
      imageAlign: 'bottom',
      imageHeight: '50px',
      imageLink: 'https://to-somewhere.go/',
      imageWidth: '50px',
      reverse: true,
      spacing: '10px',
    };

    wrapper.setProps(optionalProps);

    const updatedProps = wrapper.find(MediaImage).props();

    Object.keys(optionalProps).forEach((prop) => {
      expect(updatedProps[prop]).not.toBe(mediaImageProps[prop]);
    });
  });

  it('should have an MediaBody component', () => {
    const mediaBody = wrapper.find(MediaBody);
    expect(mediaBody).toHaveLength(1);
    mediaBodyProps = mediaBody.props();
  });

  it('that should receive props', () => {
    const newProps = {
      bodyAlign: 'bottom',
      reverse: true,
    };
    wrapper.setProps(newProps);
    const updatedProps = wrapper.find(MediaBody).props();
    Object.keys(newProps).forEach(prop => {
      expect(updatedProps[prop]).not.toBe(mediaBodyProps[prop]);
    });
  });
});