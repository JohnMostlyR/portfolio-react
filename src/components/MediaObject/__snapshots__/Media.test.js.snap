// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media should render and match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Media
    imageAlt="test"
    imageSource="/test.png"
    spacing="0"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <MediaImage
          imageAlign="top"
          imageAlt="test"
          imageHeight="auto"
          imageLink={undefined}
          imageSource="/test.png"
          imageWidth="100%"
          reverse={false}
          spacing="0"
/>,
        <MediaBody
          bodyAlign="top"
          reverse={false}
          spacing="0"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "imageAlign": "top",
          "imageAlt": "test",
          "imageHeight": "auto",
          "imageLink": undefined,
          "imageSource": "/test.png",
          "imageWidth": "100%",
          "reverse": false,
          "spacing": "0",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "bodyAlign": "top",
          "children": undefined,
          "reverse": false,
          "spacing": "0",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <MediaImage
            imageAlign="top"
            imageAlt="test"
            imageHeight="auto"
            imageLink={undefined}
            imageSource="/test.png"
            imageWidth="100%"
            reverse={false}
            spacing="0"
/>,
          <MediaBody
            bodyAlign="top"
            reverse={false}
            spacing="0"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "imageAlign": "top",
            "imageAlt": "test",
            "imageHeight": "auto",
            "imageLink": undefined,
            "imageSource": "/test.png",
            "imageWidth": "100%",
            "reverse": false,
            "spacing": "0",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "bodyAlign": "top",
            "children": undefined,
            "reverse": false,
            "spacing": "0",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
